src/abstractdataframe/sort.jl

```diff
@@ -397,14 +398,26 @@ function Base.issorted(df::AbstractDataFrame, cols=All();
                        lt::Union{Function, AbstractVector{<:Function}}=isless,
                        by::Union{Function, AbstractVector{<:Function}}=identity,
                        rev::Union{Bool, AbstractVector{Bool}}=false,
-                       order::Union{Ordering, AbstractVector{<:Ordering}}=Forward)
+                       order::Union{Ordering, AbstractVector{<:Ordering}}=Forward,
+                       checkunique::Bool=false)
     to_scalar(x::AbstractVector) = only(x)
     to_scalar(x::Any) = x
 
     # exclude AbstractVector as in that case cols can contain order(...) clauses
     if cols isa MultiColumnIndex && !(cols isa AbstractVector)
         cols = index(df)[cols]
     end
+    if checkunique
+        newcols = Int[]
+
+        for col in cols
```

Comment:
this is incorrect for two reasons:
1. `cols` might not be iterable, e.g. imagine passing `:` as column selector.
2. you drop `order` modifiers here - and these modifiers might create duplicates.

Reply:
> I haven't paid much attention to the effects of `order`, `by` and `lt`

Yes, just make sure to throw an error if user passes them and wants duplicate checking